src/Core/Backup/BackupGridDataProvider.php

```diff
@@ -0,0 +1,158 @@
+        $this->translator = $translator;
+        $this->languageDateTimeFormat = $languageDateTimeFormat;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData(SearchCriteriaInterface $searchCriteria)
+    {
+        $backups = $this->backupProvider->getBackups();
+        usort($backups, [$this, 'sortingCompare']);
+
+        $paginatedBackups = null !== $searchCriteria->getOffset() && null !== $searchCriteria->getLimit() ?
+            array_slice($backups, $searchCriteria->getOffset(), $searchCriteria->getLimit()) :
+            $backups;
+
+        $backupsArray = [];
+
+        foreach ($paginatedBackups as $backup) {
+            $backupsArray[] = [
+                'file_name' => $backup->getFileName(),
+                'file_size' => $backup->getSize(),
+                'date' => $backup->getDate(),
+                'age' => $backup->getAge(),
+                'age_formatted' => $this->getFormattedAge($backup),
+                'date_formatted' => date($this->languageDateTimeFormat, $backup->getDate()->getTimestamp()),
+                'file_size_formatted' => $this->getFormattedSize($backup),
+            ];
+        }
+
+        $backupCollection = new RowCollection($backupsArray);
+
+        return new GridData(
+            $backupCollection,
+            count($backups)
+        );
+    }
+
+    /**
+     * Compare to backup records
+     *
+     * @param BackupInterface $backup1
+     * @param BackupInterface $backup2
+     *
+     * @return bool
+     */
+    private function sortingCompare(BackupInterface $backup1, BackupInterface $backup2)
+    {
+        return $backup2->getDate()->getTimestamp() - $backup1->getDate()->getTimestamp();
+    }
+
+    /**
+     * Get formatted age
+     *
+     * @param BackupInterface $backup
+     *
+     * @return string
+     */
+    private function getFormattedAge(BackupInterface $backup)
+    {
+        if (3600 > $backup->getAge()) {
+            return sprintf('< 1 %s', $this->translator->trans('Hour', [], 'Admin.Global'));
+        }
+
+        if (86400 > $backup->getAge()) {
+            $hours = (int) floor($backup->getAge() / 3600);
+            $label = 1 === $hours ?
+                $this->translator->trans('Hour', [], 'Admin.Global') :
+                $this->translator->trans('Hours', [], 'Admin.Global');
+
+            return sprintf('%s %s', $hours, $label);
+        }
+
+        $days = (int) floor($backup->getAge() / 86400);
+        $label = 1 === $days ?
+            $this->translator->trans('Day', [], 'Admin.Global') :
+            $this->translator->trans('Days', [], 'Admin.Global');
+
+        return sprintf('%s %s', $days, $label);
+    }
+
+    /**
+     * Get formatted backup size
+     *
+     * @param BackupInterface $backup
+     *
+     * @return string
+     */
+    private function getFormattedSize(BackupInterface $backup)
+    {
+        return sprintf('%s Kb', number_format($backup->getSize() / 1000, 2));
```

Comment:
Maybe introduce size converters? as it is pretty common task. This is how i would image it:

```php
$bytes = new Bytes(1000);
$bytes->convert(Unit::MB); // converts 1000 bytes to MB
$bytes->convert(Unit::KB); // converts 1000 bytes to KB

// OR
$byteConverter = // bytes converter service
$converter->convert(1000, ConverterInterface::SIZE_MB); // converts 1000 bytes to MB
```

Reply:
you means, like [this](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Console/Helper/Helper.php#L104)?

Reply:
yup, something like this. should we implement it?

Reply:
yes, reuse it if you can: this code is already unit tested by Symfony, it's better than creating our own stuff.

Reply:
well, it does not work in our case, as we measure size in KB, MB & etc and not KiB, MiB & etc. :/

Reply:
ping @colinegin is it really important to keep Kb instead of Kib for backup size?

For the record:
* 1 Kb equals 1000 octets
* 1 Kib equals 1024 octets

Gains for us:
* we can rely on Symfony to display the information
* we can deprecate Bytes and ByteConverter classes (less code to maintain: YaY!)

Cons:
* people may not understand the difference (is it important for a backup?)
* this may change the translation key? /c @sarjon

Reply:
> people may not understand the difference (is it important for a backup?)

this would also introduce inconsistency as the rest of Back Office uses Kb, Mb & etc sizes everywhere, wouldn't that be a problem?

> this may change the translation key?

yup. :/

Reply:
Discussing with @colinegin about it, she is ok for Kib notation if we add an information notice about it to explain the difference with Kb).

She is also ok for the changes of translation key :)

Reply:
ok, but i suggest keeping it "as is" now, as it would introduce new feature. For example: Where do we put this explanation? How should it look like? Should it be reusable for other pages or specific for this one?